src/logic/messages/EventManager.h

```diff
@@ -18,7 +18,7 @@ namespace Logic
         ~EventManager();
 
         template<typename T>
-        void onMessage(const T& msg, Handle::EntityHandle sourceVob = Handle::EntityHandle::makeInvalidHandle())
+        T* onMessage(const T& msg, Handle::EntityHandle sourceVob = Handle::EntityHandle::makeInvalidHandle())
```

Comment:
I'm sorry the pull request was too early.
About the raw pointer: The memory is handled by the queue atm. If the event attribute delete is set, the event gets deleted. But I understand that returning a raw pointer can lead to many errors.
How about a ticket system? When creating a wait message, a ticket is drawn from the EventManager. In the callback, the message will be found by the ticketID (member of EventMessage) and marked as deleted (as before).